tests/update_workflow_test.go

```diff
@@ -1318,24 +1318,24 @@ func (s *UpdateWorkflowSuite) TestUpdateWorkflow_ValidateWorkerMessages() {
 				T:                   s.T(),
 			}
 
-			halfSecondTimeoutCtx, cancel := context.WithTimeout(testcore.NewContext(), 500*time.Millisecond)
+			immediateCtx, cancel := context.WithTimeout(testcore.NewContext(), 1*time.Millisecond)
```

Comment:
My hypothesis is that _occasionally_ the server responds before the client times out the request. By lowering it to 1ms, this is unlikely to happen (compared to 500ms).